cpu/cpu.go

```diff
@@ -146,12 +146,20 @@ func (p *Plugin) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, err
 // It returns error in case retrieval was not successful
 func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.MetricType, error) {
 	metrics := []plugin.MetricType{}
-	for _, metricType := range metricTypes {
-		if !p.initialized {
-			if err := p.init(metricType.Config().Table()); err != nil {
-				return nil, err
-			}
+	if len(metricTypes) == 0 {
+		return nil, nil
+	}
+	if !p.initialized {
+		if err := p.init(metricTypes[0].Config().Table()); err != nil {
+			return nil, err
 		}
+	} else {
```

Comment:
@marcin-krolik ok, I now understand what you meant and I agree with you. I did not like the fact that init was implicitly calling getStats either. However, as I did not write this part initially I did not want to change it furthermore as I might not have all background/expertise to infer all potential impacts. Just submitted a new version integrating this change. Thanks for your comments